tests/runtime/probe

```diff
@@ -464,8 +464,11 @@ PROG hardware:cache-misses:10 { print(probe); exit();}
 EXPECT hardware:cache-misses:10
 
 NAME probe_builtin_string_compare
-RUN {{BPFTRACE}} -e 'uprobe:testprogs/uprobe_test:uprobeFunction* { if (strcontains(probe, "uprobeFunction2")) { print(probe); } }' -c ./testprogs/uprobe_test
+PROG uprobe:testprogs/uprobe_test:uprobeFunction* { if (strcontains(probe, "uprobeFunction2")) { print(probe); exit(); } }
+AFTER ./testprogs/uprobe_test
 EXPECT uprobe:testprogs/uprobe_test:uprobeFunction2
+# For some reason LLVM versions older than just generate code that does not pass BPF verifier
+MIN_LLVM 18
```

Comment:
Language nit: "For some reason older LLVM versions generated code that does not pass the BPF verifier".

Larger question: isn't this a problem? You're saying strcontains will not work on llvm versions less than 18?

Reply:
The new implementation for some reason doesn't verify if built with older llvms. I plan on asking about this tomorrow. It is a problem, but a somewhat small one IMHO. The previous impl had silent wrong results + frequent LLVM crashes. The new one is correct, but under certain builds doesn't load. I'd argue new impl is strictly better b/c at least it's correct and loud if wrong.

Reply:
And we don't want to just maintain the old way for llvm versions < 18 ? When you say the previous impl had silent wrong results - do we know how often and for what cases? Maybe having something a little broken on these older LLVM version is better than nothing at all.

Reply:
Included in this PR is a runtime test that had silent wrong results for the old impl. I couldn't debug it - the old impl was too confusing. I think there's nothing worse than silent wrong results. It is much better to not do it at all than to silently do it wrong.

Reply:
With @viktormalik 's help, figured it out. I was generating bad code. Fixed to work on older LLVMs.